src/Interpreters/Context.cpp

```diff
@@ -669,13 +669,15 @@ SharedContextHolder Context::createShared()
 
 ContextMutablePtr Context::createCopy(const ContextPtr & other)
 {
+    auto lock = other->getLock();
```

Comment:
(with TSA annotations in class `Context`, such issue could (likely) be found automatically)

Reply:
Maybe yes, but there are a lot of field in `Context`. Let's do it in a separate PR.